app/src/main/java/ch/epfl/sdp/blindly/weather/WeatherActivity.kt

```diff
@@ -163,4 +190,82 @@ class WeatherActivity : AppCompatActivity(), WeatherService.WeatherResultCallbac
     override fun onRefresh() {
         weather.nextWeek(location, callback = this)
     }
-}
\ No newline at end of file
+
+    /**
+     * Forms a new Calendar Event in Google Calendar App,
+     * closes the current activity as this page is no longer needed.
+     *
+     * @param title Title of the event to add to Calendar
+     * @param date Date of the event to add to Calendar
+     */
+    private fun setCalendarEvent(title: String, date: Calendar) {
+        val calIntent = Intent(Intent.ACTION_INSERT)
+        calIntent.data = Events.CONTENT_URI
+        calIntent.putExtra(Events.TITLE, title)
+        calIntent.putExtra(CalendarContract.EXTRA_EVENT_ALL_DAY, true)
+        calIntent.putExtra(
+            CalendarContract.EXTRA_EVENT_BEGIN_TIME,
+            date.timeInMillis
+        )
+        calIntent.putExtra(
+            CalendarContract.EXTRA_EVENT_END_TIME,
```

Comment:
Call it nitpicking but I think it would be better to import all those constants to get rid of the class before them :p